stubs/django-filter/METADATA.toml

```
@@ -0,0 +1,3 @@
+version = "23.3.*"
+upstream_repository = "https://github.com/carltongibson/django-filter"
+requires = ["django-stubs", "djangorestframework-stubs"]
```

You'd need to remove this to pass the stub_uploader CI. It doesn't look like it is used anywhere in the stubs, not sure why it was included.
```suggestion
requires = ["django-stubs"]
```

---

It was used in `__init__.pyi`: `from . import rest_framework as rest_framework`. But indeed this is totally unnecessary and is the only usage, removed.

By the way, thanks a lot for improving stub-uploader, and for letting me know about the changes!

---

> It was used in `__init__.pyi`: `from . import rest_framework as rest_framework`. But indeed this is totally unnecessary and is the only usage, removed.

This looks like a locally defined module named `rest_framework`, not the djangorestframework package. I don't think it should be removed.

> By the way, thanks a lot for improving stub-uploader, and for letting me know about the changes!

Thank you and thanks for working on django stubs!

---

> This looks like a locally defined module named rest_framework

Ah right you are!

> I don't think it should be removed.

I don't think it does anything. It's re-exporting the module `django_filter.rest_framework` that is already available under the same module path? :smile:

---

> I don't think it does anything. It's re-exporting the module `django_filter.rest_framework` in `django_filter/__init__.py` that is already available under the exact same module path? 😄

If you remove it then this will not work (at least with pyright):
```python
import django_filter
django_filter.rest_framework  # <- this will be a type error and would not have autocompletion in VS Code
```

---

@intgr I still think the the removed `from . import rest_framework as rest_framework` must be restored. It is deliberately imported at runtime for the specific reason that would let pyright accept the code snippet I mentioned above, i.e. to avoid having to type `import django_fiters.rest_framework`.

See the comment here:
https://github.com/carltongibson/django-filter/blob/2494df96c6387a9fa411fcb00b696b15dfd9216b/django_filters/__init__.py#L7-L10

---

I think it's silly and nobody should use it like that. The documentation and tests don't use it that way either.

But if you feel strongly about this, fair enough.